### ISSUE TITLE
Set Link destination (within a project)

### ISSUE THREAD

**Issue by xmatthewx** [mozilla/webmaker-android#1670]
Implement the feature to allow users to pick a page within their project as the destination for their link. 

 1. In link element editor, user clicks set destination
 1. Map view appears with icon on current page
 1. User pans and taps destination page
 1. Icon appears on destination with line connecting to origin
 1. User taps checkmark in top right
 1. Return to element editor

Note
 - Unset link: When an unset link is selected in page editor, it  includes a button to set destination. 
 - Set link: When a link is selected in page editor, it includes a button to follow the link
 - Play link: In Play view, the button behaves like button. Yay.

UX is complete. UI in progress in #1549

**Comment by xmatthewx** [comment]
cc: @flukeout @vazquez for review

### Element on page

 - Button is disabled to allow select and transforms
 - Pencil opens element editor
 - `Set Link Destination` goes to page selector

![button add to page](https://cloud.githubusercontent.com/assets/1119821/7504843/26721af2-f402-11e4-89e2-be43a474027b.png)


## Link editor

 - Users can set destination and button style.
 - Tap button in preview to edit the string. 
 - Tap checkmark in header to confirm changes.
 - Nice to have: scroll overflow behind the preview.

![button options](https://cloud.githubusercontent.com/assets/1119821/7505199/5a1a1c8a-f405-11e4-869a-1ed5d026d1a4.png)

## Pick Destination Page

Users arrive here after tapping `Set Destination`

 1. Show page selector
   - center map view on the page that contains the button
   - mark this page with placemarker icon
 2. User taps destination
  - add selected state to page
  - add destination flag icon
  - confirmation checkmark appears
  - nice to have: draw connecting line between pages
 1. User confirms changes
 1. Go to element editor

![button pick destination page](https://cloud.githubusercontent.com/assets/1119821/7505333/4f7b101c-f406-11e4-8654-1e0d35715ebc.png)  

![button set destination page](https://cloud.githubusercontent.com/assets/1119821/7505335/52f8eb24-f406-11e4-9634-e995881ffa6a.png)

## Destination Set

Button helper now reads `Follow Link`

![button add to page link exists](https://cloud.githubusercontent.com/assets/1119821/7505564/cabe02d2-f408-11e4-9ebb-23ff9146f1a4.png)

**Comment by flukeout** [comment]
Looks good! What about ```Change Link Destination``` in the element editor after a page has been set.

**Comment by xmatthewx** [comment]
@Pomax - here is UI for your link destination work. Let's walk through this over vidyo before you dig into things.

**Comment by gvn** [comment]
Semi-blocked by #1903 

Also, @k88hudson wants to do some refactoring of the Project component before this gets rolling...

**Comment by gvn** [comment]
@xmatthewx Can you link me to the unflattened comps for this ticket? Thanks!

**Comment by xmatthewx** [comment]
Sure thing. The assets are updated regularly in gdrive > webmaker-android > [Version 2.0](https://drive.google.com/drive/#folders/0B-1AZgKn5Nc7fkszYTk5LTdMN2ZwdjJXb1lwM1hzTGEtc2N2ejhxSm5Pc2dkYkRUOVlWUlk/0ByIoeeW0a3R_aTF2WEx6UG1HYlE) > [Webmaker App 2.0.sketch](https://drive.google.com/open?id=0ByIoeeW0a3R_RzRuR05NdW9xU28&authuser=0). Sketch has multiple pages that you toggle in the top left. Stuff you want is in "Colour / Text options"

**Comment by gvn** [comment]
Note: This patch will leave out the tinker mode icon since that relates to another feature not yet started.

**Comment by gvn** [comment]
Created follow up tickets that should complete this feature. 

@xmatthewx @thisandagain feel free to add more tickets if I missed any details...

**Comment by xmatthewx** [comment]
:+1: I think we can call this done and file new issues as needed.
